### DISCUSSION TITLE
How to recover from error in errorElement

### DISCUSSION THREAD

**Discussion by blessanm86** [TanStack/router#224] [Q&A]
Hi would one recover from an error in the errorElement. Like is there a way to force the reload of a route?

Currently my `errorElement` looks like this.

```
import { useMatch } from "react-location";

export function ApplicationsTableBodyError() {
  const { error } = useMatch();
  return <p>Our server is having an issue - {error.message}. Try reloading the page.</p>;
}
```

I would like to put a button there and do something like `route.reload()` that only triggers the loader for that child route.

**Comment by rudzky** [comment]
Bump! That's something I also wanna know 😄

**Reply by schiller-manuel** [reply]
did you try `router.invalidate()`?

**Reply by rudzky** [reply]
I tried something like this:

```ts
export const GlobalErrorComponent = () => {
  const navigate = useNavigate();
  const router = useRouter();

  return (
    <button onClick={() => {
          navigate({
            to: "/",
          });
          router.invalidate();
        }}>Reset</button>
  )
}
```
but it doesn't work.
Let say I go to /about page, loader fails for some reason and I get this error. I guess the proper way to handle it is to go back to the homepage where loader works well and then use `invalidate` right?
